### PR TITLE
Internal: Elements - A method to sanitize settings before save [ED-7250]

### PR DESCRIPTION
**Pull request by matipojo**

### CODE REVIEW

File: includes/base/element-base.php [hunk 1]
```diff
@@ -543,6 +543,26 @@ public function get_raw_data( $with_html_content = false ) {
 		];
 	}
 
+	public function get_data_for_save() {
```
Comment (StyleShit) [inline includes/base/element-base.php:546]: it reminds me the `get_settings_for_display()` thing that nobody really knows how it works...
maybe we need a better name?

File: includes/base/element-base.php [hunk 2]
```diff
@@ -803,6 +823,14 @@ protected function get_initial_config() {
 		return $config;
 	}
 
+	/**
+	 * A Base method for sanitizing the settings before save.
+	 * This method is meant to be overridden by the element.
+	 */
+	protected function on_save( array $settings ) {
+		return $settings;
```
Comment (StyleShit) [inline includes/base/element-base.php:831]: don't you want to add some generic sanitizations?

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: includes/base/element-base.php (hunk 1 above)
Comment (matipojo) [reply includes/base/element-base.php:546]: It's OK for now

File: includes/base/element-base.php (hunk 2 above)
Comment (matipojo) [reply includes/base/element-base.php:831]: It's an infra only PR

**Pull request merged**